File: MATLAB/Erlang Noise.m
originalImage = imread('Enter your image path here');
originalImage = im2double(originalImage);

% Add Erlang noise (Gamma distribution with integer shape)
erlangShape = 2;      % k
erlangScale = 0.2;    % theta
erlangNoiseImage = originalImage;

for i = 1:size(originalImage, 3)
    erlangNoise = gamrnd(erlangShape, erlangScale, size(originalImage, 1), size(originalImage, 2));
    erlangNoiseImage(:, :, i) = originalImage(:, :, i) + erlangNoise;
end

% Clip values to [0, 1] range
erlangNoiseImage = min(max(erlangNoiseImage, 0), 1);

% Display original and noisy images
figure;
subplot(1,2,1);
imshow(originalImage);
title('Original Image');

subplot(1,2,2);
imshow(erlangNoiseImage);
title('Image with Erlang (Gamma) Noise');

File: Python/Erlang Noise.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

image = cv2.imread('Enter your image path here')
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
image = image.astype(np.float64) / 255.0

erlang_shape = 2
erlang_scale = 0.2   

erlang_noise_image = np.empty_like(image)
for i in range(3):
    erlang_noise = np.random.gamma(erlang_shape, erlang_scale, image[:, :, i].shape)
    erlang_noise_image[:, :, i] = image[:, :, i] + erlang_noise
erlang_noise_image = np.clip(erlang_noise_image, 0, 1)
plt.figure(figsize=(10, 5))
plt.subplot(1, 2, 1)
plt.imshow(image)
plt.title('Original Image')
plt.axis('off')

plt.subplot(1, 2, 2)
plt.imshow(erlang_noise_image)
plt.title('Image with Erlang Noise')
plt.axis('off')

plt.tight_layout()
plt.show()
